romm.wiki/Igir-Script.md

```diff
@@ -0,0 +1,124 @@
+# Directory structure
+
+The directory structure is important for running the bulk ROM renaming script. Your directory structure, when all data is copied in, should look something like below:
+
+```bash
+dats # DAT files downloaded from no-intro.org
+roms # your raw roms dump
+roms-unverified # a copy of roms which will be used as the input directory for the script
+igir-romm-cleanup.sh # the script to run
+```
+
+# Copy your rom files
+
+Copy your rom files to a new directory `roms-unverified`. This is useful for two reasons:
+
+1. You have confidence that anything done by the process below won't produce anything that you do not want done permanently.
+
+2. You can easily tweak the script that runs the Igir tool and rerun in the event something goes unexpectedly.
+
+# Download the latest DAT files
+
+## No-Intro
+
+Navigate to the [No-Intro.org Daily Download](https://datomatic.no-intro.org/index.php?page=download&op=daily) page. Download the latest compilation of .dat files.
+
+Create a new directory named `dats` and copy in the archive. Optionally if you only want to scan a subset you may extract a subset of .dat files into the directory instead.
+
+These DAT files work very well for cartridge consoles, but may not work well with optical redumps such as Sony PlayStation.
+
+For redumps try downloading DAT files from [redump.org](http://redump.org/downloads/) for the platforms that you are interested in.
+
+# Create the cleanup script
+
+Create a new script named `igir-romm-cleanup.sh` copying in the contents below:
+
+```bash
+#!/usr/bin/env bash
+set -ou pipefail
+cd "$(dirname "${0}")"
+
+INPUT_DIR=roms-unverified
+OUTPUT_DIR=roms-verified
+
+# Documentation: https://igir.io/
+# Uses dat files: https://datomatic.no-intro.org/index.php?page=download&op=daily
+time npx -y igir@latest \
+  move \
+  extract \
+  report \
+  test \
+  -d dats/ \
+  -i "${INPUT_DIR}/" \
+  -o "${OUTPUT_DIR}/{romm}/{language}-{region}/" \
```

Comment:
will this work? romm doesn't currently support deeply nested structures likes this. would the ideal of this script to be generate a roms folder that is compatible with romm's required structure?